.github/workflows/main.yaml

```diff
@@ -31,11 +35,34 @@ jobs:
   # Retrieves configuration from json file.
   initialize:
     runs-on: ubuntu-latest
+    if: |
```

Comment:
Valid point, but I needed to account for multiple scenarios:
- I didn't account for the `_nightly` and the `postsubmit` builds (I'll fix that).
- Run tests if the PR is ready for review.
- If the PR is `synchronized` (New changes pushed to the remote):
    - Don't run the tests in draft mode.
    - Run the tests if the PR is ready for review.
- For `runtest` label:
    - Always run the tests regardless of the PR status (draft or ready for review)
- For `on_device` label:
    - Don't do anything if the PR is in draft mode.
    - Rerun the tests again to account for on device behavior if the draft is ready for review.

I also removed the condition from the workflow yaml files to account DRY code since the condition is oddly big. I'm happy to approach differently if there's a better solution.

Reply:
I am a bit confused around the difference between 'on_device' and 'runtest' labels, what's the difference between two?  Does 'runtest' trigger presubmit without on-device tests, and 'on_device' label triggers presubmit with on-device test?

Reply:
Got the answer in our GitHub weekly meeting.